### PR TITLE
Remove nested async in sdk promise lock.

### PR DESCRIPTION
**Pull request by Hyperkid123**

### CODE REVIEW

File: packages/core/src/index.ts [hunk 1]
```diff
@@ -270,50 +271,51 @@ export async function processManifest(url: string, scope: string, module: string
...
-    // handle network errors
-    if (!response.ok) {
-      let error = 'Unable to process manifest';
-      const resClone = response.clone();
-      try {
-        error = await resClone.json();
-      } catch {
-        // try text error if json fails
-        error = await resClone.text();
+    return manifestPromise.then((response) => {
```
Comment (karelhala) [inline packages/core/src/index.ts:274]: Wouldn't it better to wrap this within `try`/`catch` clause to propagate these errors correctly?

**Review by karelhala — COMMENTED** [review]
I am kinda suprised the `async`/`await` isn't working. It should be equal to what we do right now as well. Wouldn't it be better to wrap this in `try`/`catch` and whenever an error happens call the `rej` function?

File: packages/core/src/index.ts (hunk 1 above)
Comment (Hyperkid123) [reply packages/core/src/index.ts:274]: The issue is the parent `new Promise`. That can't have an inner async function. Errors will be lost.

And yeah it also includes any inner async/await calls. So if we want to be 100% sure our inner errors are not lost, we have to do it this way.

Before, if manifest fetch resulted in 404, you would get an infinite loading state in scalprum component as the promise will not get properly resolved. Even though we were doing try/catch in the correct places.

I don't like this either but I don't think we have any other option.

**Comment by Hyperkid123** [comment]
The async/await alternative will be addressed in a follow up

**Pull request merged**